python/pyarrow/parquet.py

```diff
@@ -231,6 +231,8 @@ class ParquetWriter(object):
 {0}
 """.format(_parquet_writer_arg_docs)
 
+    is_open = False
```

Comment:
I don't think this is the correct solution. It looks like we're defining `__del__` which will be called if `__init__` raises an exception. If `__init__` raises an exception *before* defining `self.is_open` then you'll get an `AttributeError` exception in `__del__` when it's called.

We really should *not* be opening files in constructors, since Python doesn't have the same guarantees as C++ around resource initialization and destruction. We should be using a context manager here since that provides clear semantics and guarantees when `__enter__` and `__exit__` will run.

For now, though, can you do this instead?

```python
def __del__(self):
    if getattr(self, 'is_open', False):
        self.close()
```